### DISCUSSION TITLE
Saving data to qualtrics

### DISCUSSION THREAD

**Discussion by advale123** [jspsych/jsPsych#867] [Q&A] [answered]
Hi
I created my experiment using jspsych, and hosted it on a google server (google firebase).
Initially I planned to save participants' data to the storage google provide on those servers, but it is costly and doesn't work well with the platform I use for recruiting (Prolific), so I figured I can embed my link to a qulatrics survey - It seems fine, I just can't figure out how to save the data to qualtrics.
Anyone has experience with embedding jspsych code in a qualtrics survay?

**Comment by jodeleeuw** [comment, marked as answer]
I don't have experience with Qualtrics data saving. Another option is www.cognition.run. I recorded a tutorial video showing how to integrate with Prolific:

https://www.youtube.com/watch?v=O628dNA7WCc

**Reply by advale123** [reply]
Thank you for this tutorial! I think it is exactly what we were looking for :)

do you know by any chance what is the storage limitation on cognition?

**Reply by jodeleeuw** [reply]
I'm not sure.  Lots of Q&A about cognition.run has happened in this thread: https://github.com/jspsych/jsPsych/discussions/754

**Comment by vijaymarupudi** [comment]
I have made a library in the past for this before: <https://gitlab.com/vijaymarupudi/qualutils>

Feel free to give it a try. I'm not willing to support it though, Qualtrics has been very hostile to developers in the past, so I'd recommend alternatives if possible.

**Comment by jkhartshorne** [comment]
@advale123 - I'd like to back up: why are you including a jsPsych experiment in a link from a Qualtrics survey? Is Qualtrics necessary? Why do you need it?

**Reply by advale123** [reply]
Qualtrics is not necessary, I originally hosted my experiment on google Firebase and planned to use it for saving my data, but it seemed like it would be quit costly and we already paid for a qualtrics account so I figured I could use it. 
I would be happy to use any other platform you suggest, as long as I can use it for free and connect it to Prolific.
